rust/lance-core/Cargo.toml

```diff
@@ -23,8 +23,6 @@ arrow-select.workspace = true
 async-recursion.workspace = true
 async-trait.workspace = true
 lance-arrow.workspace = true
-aws-config.workspace = true
-aws-credential-types.workspace = true
 aws-sdk-dynamodb = { workspace = true, optional = true }
```

Comment:
I suspect there are more dependencies that can be removed.  I want to take a pass at doing this later.